File: tests/previous_high_low_test.py
import numpy as np

from taflow import PreviousHighLow


def test_previous_high_low_lifecycle_and_reset():
    state = PreviousHighLow(np.array([], dtype=bool), np.array([], dtype=float), np.array([], dtype=float))
    state.extend([True, True], [10.0, 12.0], [8.0, 7.0])
    assert state.compute()[0][-1] == 10.0
    assert len(state) == 2
    state.reset()
    assert state.value is None

File: crates/taflow-core/src/indicators/previous_high_low_test.rs
use super::previous_high_low::PreviousHighLow;

#[test]
fn session_breaks_and_reset_are_consistent() {
    let mut state = PreviousHighLow::new();
    state.append(true, 10.0, 8.0);
    let value = state.append(true, 12.0, 7.0);
    assert_eq!(value.prev_high, 10.0);
    state.reset();
    assert!(state.value().is_none());
}
